### PR TITLE
ISSUE-404: Support http proxy related configuration for registry server

### PR DESCRIPTION
**Pull request by priyank5485**
Support http proxy related configuration for registry server

### CODE REVIEW

**Comment by priyank5485** [comment]
@arunmahadevan That is correct. Yes, ultimately the goal is to have only one set of configs at top level. For that, we need a change in streamline code to pick those configs up from top level. This change in registry seemed independent and was to fix the proxy issue related to downloading mysql jar in TablesInitializer call, that was being used in streamline and registry bootstrap-storage.sh call. I think we can independently merge this and raise a separate PR to handle streamline change to avoid that config at two levels. Let me know if you think otherwise

**Comment by priyank5485** [comment]
I have also added those 3 fields to RegistryConfiguration and squashed and pushed my branch.

**Comment by sapinamin** [comment]
I have submitted the ambari PR https://github.com/hortonworks/hdf_ambari_mp/pull/187 and tested it with jar with this PR. Please merge it.

**Comment by arunmahadevan** [comment]
Are the changes proposed in this PR still required. Does https://github.com/hortonworks/streamline/pull/1215 override this ?

**Comment by arunmahadevan** [comment]
@priyank5485 , ignore the previous comment. I didnt realize this patch is for registry and the other one is for streamline.

So the question is: does https://github.com/hortonworks/streamline/pull/1215 depend on the changes proposed in this patch?

File: conf/registry-oracle-example.yaml [hunk 1]
```diff
@@ -67,6 +67,11 @@ swagger:
 #enable CORS, may want to disable in production
 enableCors: false
 
+#Set below 3 properties if server needs a proxy to connect. Useful to download mysql jar
```
Comment (arunmahadevan) [inline conf/registry-oracle-example.yaml]: nit : i dont think we should reference mysql in registry-oracle-example.yaml

File: conf/registry-inmemory-example.yaml [hunk 2]
```diff
@@ -60,6 +60,11 @@ swagger:
 #enable CORS, may want to disable in production
 enableCors: false
 
+#Set below 3 properties if server needs a proxy to connect. Useful to download mysql jar
```
Comment (arunmahadevan) [inline conf/registry-inmemory-example.yaml]: nit : i dont think we should reference mysql in this file

File: conf/registry-postgres-example.yaml [hunk 3]
```diff
@@ -67,6 +67,11 @@ swagger:
 #enable CORS, may want to disable in production
 enableCors: false
 
+#Set below 3 properties if server needs a proxy to connect. Useful to download mysql jar
```
Comment (arunmahadevan) [inline conf/registry-postgres-example.yaml]: nit : i dont think we should reference mysql in this file

File: conf/registry-oracle-example.yaml (hunk 1 above)
Comment (priyank5485) [reply conf/registry-oracle-example.yaml]: Addressed

File: conf/registry-inmemory-example.yaml (hunk 2 above)
Comment (priyank5485) [reply conf/registry-inmemory-example.yaml]: Addressed

File: conf/registry-postgres-example.yaml (hunk 3 above)
Comment (priyank5485) [reply conf/registry-postgres-example.yaml]: Addressed

**Comment by priyank5485** [comment]
@arunmahadevan I have addressed your comments and removed mysql references in non mysql yaml files. Streamline PR that you asked about depends on this PR for downloading mysql jar during setup.

**Comment by arunmahadevan** [comment]
+1

**Pull request merged**